tests/test_oml/test_datasets/test_audio.py

```diff
@@ -0,0 +1,95 @@
+import random
+
+import pandas as pd
+import pytest
+
+from oml.const import PATHS_COLUMN, START_TIME_COLUMN
+from oml.utils import get_mock_audios_dataset
+from tests.test_oml.test_datasets.test_datasets import (
+    check_base,
+    check_labeled,
+    check_query_gallery,
+)
+
+
+@pytest.fixture(scope="module")
+def df() -> pd.DataFrame:
+    df_train, df_val = get_mock_audios_dataset(global_paths=True)
+    return pd.concat([df_train, df_val])
+
+
+@pytest.mark.needs_optional_dependency
+@pytest.mark.parametrize("num_channels", [1, 2])
+def test_downmix(df: pd.DataFrame, num_channels: int) -> None:
+    from oml.datasets.audios import AudioBaseDataset
+
+    dataset = AudioBaseDataset(df[PATHS_COLUMN].tolist(), num_channels=num_channels)
+    for item in dataset:
+        audio = item[dataset.input_tensors_key]
+        assert audio.shape[0] <= num_channels, f"Audio channels {audio.shape[0]} exceed specified {num_channels}"
+
+
+@pytest.mark.needs_optional_dependency
+@pytest.mark.parametrize("sr", [8000, 16000, 22050, 44100, 48000])
+@pytest.mark.parametrize("max_num_seconds", [0.01, 0.5, 1.0, 3.0, 10.0, 100.0])
+def test_resample_trim_pad(df: pd.DataFrame, sr: int, max_num_seconds: float) -> None:
+    from oml.datasets.audios import AudioBaseDataset
+
+    dataset = AudioBaseDataset(df[PATHS_COLUMN].tolist(), sr=sr, max_num_seconds=max_num_seconds)
+    for item in dataset:
+        audio = item[dataset.input_tensors_key]
+        assert audio.shape[1] == int(
+            max_num_seconds * sr
+        ), f"Audio length {audio.shape[1]} does not match expected {int(max_num_seconds * sr)}"
+
+
+@pytest.mark.needs_optional_dependency
+def test_start_times(df: pd.DataFrame) -> None:
+    from oml.datasets.audios import AudioBaseDataset, AudioLabeledDataset
+
+    # test base dataset
+    start_times = [random.uniform(0, 1) for _ in range(len(df))]
```

Comment:
1. the value can be greater than 1 because it's number of seconds
2. let's add a few empty values as well as I suggested above

Reply:
fixed